src/poetry/vcs/git/backend.py

```diff
@@ -313,21 +314,38 @@ def _clone(cls, url: str, refspec: GitRefSpec, target: Path) -> Repo:
             if isinstance(e, AssertionError) and "Invalid object name" not in str(e):
                 raise
 
-            logger.debug(
-                "\nRequested ref (<c2>%s</c2>) was not fetched to local copy and"
-                " cannot be used. The following error was"
-                " raised:\n\n\t<warning>%s</>",
-                refspec.key,
-                e,
-            )
+            short_ref_found = False
 
-            raise PoetryConsoleError(
-                f"Failed to clone {url} at '{refspec.key}', verify ref exists on"
-                " remote."
-            )
+            if refspec.is_sha_short:
```

Comment:
@mikhainin can we move the new logic into an `else` block under the for loop here;

https://github.com/python-poetry/poetry/blob/6f6fd7012983a2e749c6030c1f5f155fd4397058/src/poetry/vcs/git/backend.py#L95-L101

And as @jelmer suggested, we can simplify it to something like 

```py
            else:
                for sha in repo.object_store.iter_prefix(prefix):
                    self.revision = sha.decode("utf-8")
                    break
```

Assuming we get a dulwich release with the fix soon.

Reply:
I'll see if I can do a release this weekend,

Reply:
0.22.6 is now out with the fix

Reply:
No it isn't!  I guess your release pipeline failed?

Reply:
Right, I moved the logic to `_normalise()` - it now looks even cleaner.
Waiting for **0.22.6** to appear and then will push the change.

Reply:
Dulwich 0.22.6 is out